### PR TITLE
Update dependencies in quantum chemistry tutorial

### PR DESCRIPTION
**Pull request by soranjh**
This PR updates the quantum chemistry tutorial to make it consistent with the quantum chemistry unification in PennyLane.

### CODE REVIEW

File: demonstrations/tutorial_quantum_chemistry.py [hunk 1]
```diff
@@ -149,20 +135,9 @@
...
-# results of the mean field calculation. Then, it builds the fermionic Hamiltonian and
-# maps it to the qubit representation.
-
-qubit_hamiltonian = qchem.decompose(hf_file, mapping="jordan_wigner")
-print("Qubit Hamiltonian of the water molecule")
-print(qubit_hamiltonian)
 
 ##############################################################################
-# It is often convenient to use the :func:`~.pennylane_qchem.qchem.molecular_hamiltonian`
+# It PennyLane we have the :func:`~.pennylane.qchem.molecular_hamiltonian`
```
Comment (ixfoduap) [inline demonstrations/tutorial_quantum_chemistry.py:140]:
```suggestion
# In PennyLane we have the :func:`~.pennylane.qchem.molecular_hamiltonian`
```

File: demonstrations/tutorial_quantum_chemistry.py [hunk 2]
```diff
@@ -278,6 +248,32 @@
 # observable is an approximation of the Hamiltonian generated in the
 # section :ref:`hamiltonian`.
 #
+# OpenFermion-PySCF backend
+# -------------------------
+# The :func:`~.pennylane.qchem.molecular_hamiltonian` function can be also used to construct the
+# molecular Hamiltonian with a non-differentiable backend that uses the
+# `OpenFermion-PySCF <https://github.com/quantumlib/OpenFermion-PySCF>`_ plugin interfaced with the
+# electronic structure package `PySCF <https://github.com/sunqm/pyscf>`_. The non-differentiable
+# backend can be selected by setting `method='pyscf'` in
```
Comment (ixfoduap) [inline demonstrations/tutorial_quantum_chemistry.py:257]:
```suggestion
# electronic structure package `PySCF <https://github.com/sunqm/pyscf>`_. This
# backend can be selected by setting `method='pyscf'` in
```
I don't think it's helpful to constantly emphasize that this method is not differentiable. Makes us sound like we are down playing it

File: demonstrations/tutorial_quantum_chemistry.py [hunk 3]
```diff
@@ -278,6 +248,32 @@
...
+# electronic structure package `PySCF <https://github.com/sunqm/pyscf>`_. The non-differentiable
+# backend can be selected by setting `method='pyscf'` in
+# :func:`~.pennylane.qchem.molecular_hamiltonian`:
+
+symbols = ["H", "H"]
+geometry = np.array([[0.0, 0.0, 0.0], [0.0, 0.0, 2.0]])
+H, qubits = qchem.molecular_hamiltonian(symbols, geometry, method='pyscf')
+
+##############################################################################
+# The non-differentiable backend requires the ``OpenFermion-PySCF`` plugin to be installed by the
```
Comment (ixfoduap) [inline demonstrations/tutorial_quantum_chemistry.py:265]:
```suggestion
# This backend requires the ``OpenFermion-PySCF`` plugin to be installed by the
```

**Review by ixfoduap — APPROVED** [review]
Approving pre-emptively in the interest of time, but please implement my suggested changes first

File: demonstrations/tutorial_quantum_chemistry.py (hunk 1 above)
Comment (agran2018) [inline demonstrations/tutorial_quantum_chemistry.py:140]:
```suggestion
# In PennyLane we have the :func:`~.pennylane.qchem.molecular_hamiltonian`
```

File: demonstrations/tutorial_quantum_chemistry.py [hunk 4]
```diff
@@ -278,6 +248,32 @@
...
+# The :func:`~.pennylane.qchem.molecular_hamiltonian` function can be also used to construct the
+# molecular Hamiltonian with a non-differentiable backend that uses the
+# `OpenFermion-PySCF <https://github.com/quantumlib/OpenFermion-PySCF>`_ plugin interfaced with the
+# electronic structure package `PySCF <https://github.com/sunqm/pyscf>`_. The non-differentiable
+# backend can be selected by setting `method='pyscf'` in
+# :func:`~.pennylane.qchem.molecular_hamiltonian`:
+
+symbols = ["H", "H"]
+geometry = np.array([[0.0, 0.0, 0.0], [0.0, 0.0, 2.0]])
+H, qubits = qchem.molecular_hamiltonian(symbols, geometry, method='pyscf')
```
Comment (agran2018) [inline demonstrations/tutorial_quantum_chemistry.py:262]: Weird print out her
Screenshot from 2022-04-22 10-53-50
e

File: demonstrations/tutorial_quantum_chemistry.py [hunk 5]
```diff
@@ -278,6 +248,32 @@
...
+
+##############################################################################
+# The non-differentiable backend requires the ``OpenFermion-PySCF`` plugin to be installed by the
+# user with
+#
+# .. code-block:: bash
+#
+# pip install openfermionpyscf
```
Comment (agran2018) [inline demonstrations/tutorial_quantum_chemistry.py:270]: Perhaps this fixes the empty box above?
```suggestion
#     pip install openfermionpyscf
```

**Review by agran2018 — CHANGES_REQUESTED** [review]
Thank you @soranjh !. Select "Request changes" as there are two things that do not render OK.

**Review by agran2018 — APPROVED** [review]
Thanks @soran!

**Pull request merged**